src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java

```diff
@@ -388,6 +389,14 @@ public final void copy(Receiver receiver, int index, Location location) {
         throw createUnsupportedOperationException();
     }
 
+    /**
+     * Get Namespace URI. Throws {@code UnsupportedOperationException}.
+     */
+    @Override
+    public NamespaceUri getNamespaceUri() {
```

Comment:
This is the only thing I am unsure on how to do, so I made it unsupported until it is figured out.

Reply:
@timurt Are you able to just clarify its not need it or help us create code for it?

Reply:
This code is needed.

````
Caused by: java.lang.UnsupportedOperationException: Operation is not supported
	at com.puppycrawl.tools.checkstyle.xpath.AbstractNode.createUnsupportedOperationException(AbstractNode.java:407)
	at com.puppycrawl.tools.checkstyle.xpath.AbstractNode.getNamespaceUri(AbstractNode.java:397)
	at net.sf.saxon.pattern.NameTest.test(NameTest.java:181)
	at net.sf.saxon.tree.util.Navigator$AxisFilter.next(Navigator.java:1250)
	at net.sf.saxon.tree.util.Navigator$AxisFilter.next(Navigator.java:1223)
	at net.sf.saxon.expr.MappingIterator.next(MappingIterator.java:78)
	at net.sf.saxon.om.FocusTrackingIterator.next(FocusTrackingIterator.java:72)
	at net.sf.saxon.expr.FilterIterator.getNextMatchingItem(FilterIterator.java:79)
	at net.sf.saxon.expr.FilterIterator.next(FilterIterator.java:63)
	at net.sf.saxon.om.SequenceTool.supply(SequenceTool.java:145)
	at net.sf.saxon.expr.sort.DocumentOrderIterator.<init>(DocumentOrderIterator.java:41)
	at net.sf.saxon.expr.sort.DocumentSorter.iterate(DocumentSorter.java:276)
	at net.sf.saxon.sxpath.XPathExpression.evaluate(XPathExpression.java:192)
	at com.puppycrawl.tools.checkstyle.checks.coding.MatchXpathCheck.findMatchingNodesByXpathQuery(MatchXpathCheck.java:291)
	at com.puppycrawl.tools.checkstyle.checks.coding.MatchXpathCheck.beginTree(MatchXpathCheck.java:274)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyBegin(TreeWalker.java:299)
````

Reply:
I found `NamespaceUri.NULL` in `NameTest` so I am going with that for now as it seems to work.